crates/infra/cli/src/toolchains/napi/cli.rs

```diff
@@ -114,3 +119,87 @@ impl NapiCli {
             .run();
     }
 }
+
+#[cfg(target_env = "gnu")]
+/// On a GNU host, cross-compile the native addon to only target the oldest supported GLIBC version by VS Code.
+///
+/// By default, compiling on the host targets the host's GLIBC version, which is usually newer.
+/// To prevent that, we need to explicitly cross-compile for the desired GLIBC version.
+///
+/// This is necessary to retain extension compatibility with as many systems as possible:
+/// <https://code.visualstudio.com/docs/supporting/requirements#_additional-linux-requirements>.
+fn ensure_correct_glibc_for_vscode(
+    resolver: &NapiResolver,
+    output_dir: &Path,
+    target: &BuildTarget,
+) -> Result<()> {
+    let target_triple = match target {
+        BuildTarget::ReleaseTarget(target) if target.ends_with("-linux-gnu") => target,
+        _ => return Ok(()),
+    };
+
+    let min_glibc = NapiConfig::target_glibc(resolver)?;
+
+    let output_artifact = match target_triple.split('-').next() {
```

Comment:
nit: should we extract this to a util? target triple to bindings file name?

Reply:
`compile_platform_packages` gets the napi-style triple (e.g. `linux-x64-gnu`) from the file system as part of the napi's CLI output, so this'd mean we'd have to hardcode this enumeration ourselves and the only place where it'd be useful is here, which is more of a hack/workaround than anything until the napi CLI v3 is released proper.

In principle I agree but I think it's more effort than it's worth it, to be honest; further dumbing it down to just `addon_file_name = |platform: &str| => format!("index.{platform}.node") seems less useful as it seems like an abstraction that obscures rather than helps understand what's going on.

I'd rather we just get rid of this code entirely once CLI v3 is published =)

Reply:
Sounds good.